src/ui-kit/experimental/header/header.component.ts

```diff
@@ -0,0 +1,100 @@
+import { Component, Input } from '@angular/core';
+import { HeaderModel, HeaderNavigationLink, HeaderSecondaryLink } from './model/HeaderModel';
+
+
+@Component({
+  selector: 'sam-header-v2',
+  templateUrl: './header.component.html',
+  styleUrls: ['./header.component.scss']
+})
+export class SamHeaderComponent {
+
+  /**
+   * Model used for the different display portions of the header 
+   */
+  @Input() model: HeaderModel;
+
+  /**
+   * Takes in a text string and removes all white space characters and returns the new string
+   * @param text 
+   */
+  removeWhiteSpace(text: string) {
+    return text.replace(/ /g, '');
+  }
+
+  /**
+   * Deselects previous seletion
+   * @param id 
+   */
+  select(id: string) {
+    this.deselect();
+    let item = this.find(id);
+    if (item) {
+      item.selected = true;
+    }
+  }
+
+  /**
+   * Deselects all the items in the header model
+   */
+  deselect() {
+    if (this.model) {
+      if (this.model.home) {
+        this.model.home.selected = false;
+      }
+      if (this.model.navigationLinks) {
+        this.model.navigationLinks.forEach(function (item: HeaderNavigationLink) {
+          item.selected = false;
+          if (item.children) {
+            item.children.forEach(function (child: HeaderNavigationLink) {
+              child.selected = false;
+            });
+          }
+        });
+      }
+      if (this.model.secondaryLinks) {
+        this.model.secondaryLinks.forEach(function (item: HeaderSecondaryLink) {
+          item.selected = false;
+        });
+      }
+    }
+  }
+
+  /**
+   * Finds the navigation element by id in the header model 
+   * @param id of the navigation item
+   */
+  find(id: string) {
```

Comment:
@thomastighe Overall, this is a pretty large function. It would be better to try to break this up. There are lots of cases in here, so your tests will have to pass in a lot of cases to cover all the branches.